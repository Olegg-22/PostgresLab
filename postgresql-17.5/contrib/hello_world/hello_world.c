#include "postgres.h"
#include "fmgr.h"
#include "utils/builtins.h"

PG_MODULE_MAGIC;

Datum hello_world_c(PG_FUNCTION_ARGS);
PG_FUNCTION_INFO_V1(hello_world_c);

Datum
hello_world_c(PG_FUNCTION_ARGS)
{
    text *input = PG_GETARG_TEXT_P(0);
    char *input_str = text_to_cstring(input);
    char *output_str = (char *) palloc(strlen(input_str) + 7+7 + 1); // "Hello from c ," + input + '\0'

    sprintf(output_str, "Hello from c, %s", input_str);
    PG_RETURN_TEXT_P(cstring_to_text(output_str));
}
