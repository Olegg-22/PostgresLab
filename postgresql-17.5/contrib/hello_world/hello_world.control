comment = 'Пример расширения hello_world'
module_pathname = '$libdir/hello_world'
default_version = '1.0'
