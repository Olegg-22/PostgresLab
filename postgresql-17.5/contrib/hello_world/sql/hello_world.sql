CREATE EXTENSION hello_world;

SELECT hello_world_c('world');
SELECT hello_world_c('Abrick');
SELECT hello_world('world');
