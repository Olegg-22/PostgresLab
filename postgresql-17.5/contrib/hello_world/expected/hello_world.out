CREATE EXTENSION hello_world;
SELECT hello_world_c('world');
    hello_world_c    
---------------------
 Hello from c, world
(1 row)

SELECT hello_world_c('Abrick');
    hello_world_c     
----------------------
 Hello from c, Abrick
(1 row)

SELECT hello_world('world');
 hello_world  
--------------
 Hello, world
(1 row)

